### PR TITLE
With --compression auto,C, only use C if lz4 achieves at least 3% compression

### PR DESCRIPTION
**Pull request by jdchristensen**

### CODE REVIEW

File: src/borg/helpers.py [hunk 1]
```diff
@@ -2162,13 +2162,13 @@ def heuristic_lz4(self, compr_args, chunk):
         cdata = lz4.compress(data)
         data_len = len(data)
         cdata_len = len(cdata)
-        if cdata_len < data_len:
+        if cdata_len < 0.97*data_len:
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: let's see if that triggers the pep8 checker. usually, spaces around operators are desired.

Comment (ThomasWaldmann) [inline src/borg/helpers.py]: no, it didn't.

Comment (jdchristensen) [inline src/borg/helpers.py]: Want me to change it, regardless of what the checker says?

Comment (enkore) [inline src/borg/helpers.py]: If you don't mind ;)

Comment (jdchristensen) [inline src/borg/helpers.py]: I went ahead and amended the commit to use spaces there.

**Comment by ThomasWaldmann** [comment]
another idea that just came to me:

this change now somehow ignores possible compression ratios in the [0.97 ..1.0) - even if we just have proven that even lz4 can achieve them.

if we consider it is worth it, we could decide on lz4 compression for e.g. the [0.97 .. 0.99] range.

in a later optimization, we can even reuse the lz4 compression results so we do not need to compress again.

**Comment by enkore** [comment]
I think that would tie nicely into making CompressionDecider2 a borg.compress.Compressor implementation.

**Comment by jdchristensen** [comment]
The above ideas are nice, but probably only worth it if the code doesn't become too complicated.  Maybe best to postpone for now?

Are those build errors anything to worry about?

**Comment by ThomasWaldmann** [comment]
@jdchristensen the only change in question / in scope of this PR is adding another 2 lines like (pseudo code):
```
...
elif size < orig * 0.99:
    compr = 'lz4'
...
```

**Comment by ThomasWaldmann** [comment]
travis is happy, ignore codecov.

**Comment by jdchristensen** [comment]
I don't think it's worth redoing the lz4 compression for <= 2%.  It also doesn't follow the current documentation, and I don't think it's worth making the compression documentation even more complicated.

**Comment by ThomasWaldmann** [comment]
OK, thanks for the PR!

**Pull request merged**